netket/graph/lattice.py

```diff
@@ -19,9 +19,17 @@
 import itertools
 import networkx as _nx
 import warnings
+from typing import Tuple, Union, Optional
+
+cutoff_tol = 1e-5
```

Comment:
I think this should be a parameter that is passed as an option at creation time of ```Lattice``` (with a default value of 1.0e-5) in other words, we should leave the possibility to change it to the user

Reply:
I'd keep this as is? 
This is really only a floating-point precision error, I don't see it creating issues in general...

Reply:
```numpy.isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False)```

I would something like this, passing expliciitly rtol as a parameter. Some lattices might require smaller rtol than our (rather arbitrary chosen) 1.0e-5... the main issue is that in ```sparse_distance_matrix``` we are effectively fixing the atol=1.0e-5, which I think can be rather problematic....

Reply:
ok i agree

Reply:
Let's put it as `distance_atol`, a member in the class and pass it around.

Reply:
I think this is an absolute tolerance rather than a relative?

Reply:
yes this is an absolute tolerance, unfortunately we can only use an absolute one since ```cKDTree``` does not allow us to pass relative tolerances

Reply:
Ok, so I will define `distance_atol` as an argument of the init method for the `Lattice` with a 1e-5 default value, and pass it to the function that uses the `cKDTree`